### PR TITLE
Update NodePaths only in built-in resources

### LINKED ISSUE #84910: Reparenting nodes is extremely slow in certain situations
### Godot version

4.2 beta6

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

https://github.com/godotengine/godot/assets/2223172/8f2c5b00-f440-4403-9a62-07316a63cc10

The culprit is one of the recursive check PRs. I profiled the editor and in the case above most of the time is spent in `get_property_list()` method of TileAtlasSource, called by `_check_node_path_recursive()` in SceneTreeDock. Apparently this tile method is very slow and here it's called tens of times.

Not sure what the fix would be, maybe the editor should skip external resources (is it safe to assume they don't point to anything on the scene?). Or we could have a list of exceptions that are known to be slow and don't have NodePaths.

### Steps to reproduce

1. Open the MRP
2. Move Node2D under TileMap

### Minimal reproduction project

[LagSet.zip](https://github.com/godotengine/godot/files/13355862/LagSet.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #84910

### CODE REVIEW

**Comment by Mickeon** [comment]
It's been a while, could use another rebase?

**Comment by KoBeWi** [comment]
Why? There are no conflicts and the code is unlikely to have stopped working.

**Review by Mickeon — APPROVED** [review]
🤷 I myself think this is a good idea then.

**Comment by KoBeWi** [comment]
Note #85517, which "fixes" the same issue by disabling functionality.

Also relevant discussion from chat.
image
(pasting as image, because it's from private channel)

**Comment by Mickeon** [comment]
My opinion is, I think the functionality should be restored but limited as this PR does. Checking only for built-in resources' NodePaths is not just performant, but much, much safer as it pretty much ensures the updated NodePath points to a valid node in the scene. If it also affected external Resources there would be no way of guaranteeing it.

**Comment by KoBeWi** [comment]
Rebased and added revert.

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -1843,7 +1846,6 @@ bool SceneTreeDock::_check_node_path_recursive(Node *p_root_node, Variant &r_var
 			}
 			break;
 		};
```
Comment (akien-mga) [inline editor/scene_tree_dock.cpp:1848]: Just a consistency nitpick.
```suggestion
		} break;
```

**Review by akien-mga — APPROVED** [review]
Looks good to me, I think we can squash the commits.

**Comment by akien-mga** [comment]
Thanks!

**Comment by ShadyShlowpoke** [comment]
I'm on Godot 4.3 (beta1) and while in the editor adding/moving/removing nodes is still painfully slow when many nodes are in the scene. Is this still being worked on?

**Comment by KoBeWi** [comment]
The issue linked to this pull request is closed. If the problem still exists in some form, see if it was already reported and open a new issue if it wasn't.

**Comment by ShadyShlowpoke** [comment]
This may be related to #83460, which is still open. Editor freezes when adding, moving, or removing nodes in a scene with many nodes, even if they are just built-in nodes like Node2Ds. The more nodes, the longer the freeze.

**Pull request merged**